### ISSUE TITLE
Some specific inputs for the page.get_pixmap() method causes the program to freeze.

### ISSUE THREAD

**Issue by myhloli** [pymupdf/PyMuPDF#3072]
### Description of the bug

[zlib_21929367.pdf](https://github.com/pymupdf/PyMuPDF/files/14004330/zlib_21929367.pdf)
when i use page.get_pixmap() to cut image in this pdf（page_id=48,page_id=49,page_id start from 0),the program will freeze.

### How to reproduce the bug

simple_1
```
doc = fitz.open(pdf_file)
page_48 = doc[48]
img_bbox = [147, 300, 447, 699]
rect = fitz.Rect(*bbox)
zoom = fitz.Matrix(3, 3)
pix = page_48 .get_pixmap(clip=rect, matrix=zoom)
image_save_path = "1.jpg"
pix.save(image_save_path, jpg_quality=95)
```
simple_2
```
doc = fitz.open(pdf_file)
page_49 = doc[49]
img_bbox = [147, 543, 447, 768]
rect = fitz.Rect(*bbox)
zoom = fitz.Matrix(3, 3)
pix = page_49 .get_pixmap(clip=rect, matrix=zoom)
image_save_path = "2.jpg"
pix.save(image_save_path, jpg_quality=95)
```

### PyMuPDF version

1.23.16

### Operating system

Windows

### Python version

3.10

**Comment by JorjMcKie** [comment]
This is weird:
The full page pixmap can be created and saved. Specifying clips around the images however leads to loops in `get_pixmap()`.

**Comment by myhloli** [comment]
> This is weird: The full page pixmap can be created and saved. Specifying clips around the images however leads to loops in `get_pixmap()`.

you are right ,making minor adjustments to the coordinate offset of the bbox by a few pts may can allow the program to execute normally.
for example，set simple_1.bbox = [147, 300, 448, 699] or simple_1.bbox = [147, 299, 447, 699], program will run normally；
but set simple_1.bbox = [148, 300, 448, 699] or simple_1.bbox = [147, 300, 448, 700] can't work.

**Comment by JorjMcKie** [comment]
Exactly! This works:
```python
import fitz
import os

scriptdir = os.path.dirname(__file__)
delta = (0, -0.01, 0, 0)
doc = fitz.open(os.path.join(scriptdir, "test.pdf"))
page_48 = doc[48]
rect = fitz.Rect(147, 300, 447, 699) + delta
zoom = fitz.Matrix(3, 3)
pix = page_48.get_pixmap(clip=rect, matrix=zoom)
image_save_path = os.path.join(scriptdir, "1.jpg")
pix.save(image_save_path, jpg_quality=95)

page_49 = doc[49]
rect = fitz.Rect(147, 543, 447, 768) + delta
pix = page_49.get_pixmap(clip=rect, matrix=zoom)
image_save_path = os.path.join(scriptdir, "2.jpg")
pix.save(image_save_path, jpg_quality=95)
```
Weird!!!

**Comment by myhloli** [comment]
if bbox = [x0, y0, x1, y1]
delta change y0 or x1 make program work，
delta change x0 or y1 then it can‘t run。

**Comment by JorjMcKie** [comment]
It is a weird phenomenon. Need some time to dig into it and consult with colleagues.
BTW you are aware that you can extract the images directly - without taking a pixmap?
You seem to be interested in them specifically.
Just use `doc.extract_image(xref)` with the xref of the image. You will get the original image, including all properties like resolution, etc.
Specifying a large resolution for the page pixmap will not improve the images in anyway. On the contrary: what you get in this way will likely have some degradation in one aspect or another.

**Comment by myhloli** [comment]
Indeed, I understand that images can be directly extracted, but in the process of implementing PDF parsing, there are instances where small images overlap on larger images, or text blocks overlap on large images, or vector graphics overlap on pixel images. Therefore, I opted to render the entire page and then cut pixel images from specific areas. This ensures that the reproduced document closely resembles the display of the original PDF.
Thank you for your answer.

**Comment by julian-smith-artifex-com** [comment]
There's a fix in mupdf for this. The next release of pymupdf will use the new mupdf, so marking this as fixed in next release.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.22.
